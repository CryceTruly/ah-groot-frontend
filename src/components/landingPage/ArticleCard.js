import React, { Component } from 'react';
import moment from 'moment';

class ArticleCard extends Component {
  render() {
    const { title, description, body, reading_time, updated_at } = this.props;
    const newdate = new Date(updated_at).toString();
    const timeAgo = moment(`${newdate}`).fromNow();

    return (
      <div className='col s6'>
      <div className="card grey lighten-5">
        <div className="card-content black-text">
          <span className="card-title truncate">{title}</span>
          <span className="card-description truncate">{description}</span>
          <div className="truncate">{body}</div>
        </div>
        <div className="card-action">
          <div>published {timeAgo}</div>
          <div>{reading_time}</div>
          <i className="material-icons">star</i>
        </div>
      </div>
      </div>
    );
  }
}

export default ArticleCard;
