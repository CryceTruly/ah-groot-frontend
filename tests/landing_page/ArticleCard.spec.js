import React from 'react';
import { shallow } from 'enzyme';

import ArticleCard from '../../src/components/landingPage/ArticleCard';

describe('ArticleCard', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<ArticleCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
