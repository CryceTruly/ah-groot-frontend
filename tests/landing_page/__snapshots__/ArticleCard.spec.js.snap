// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard should render without crashing 1`] = `ShallowWrapper {}`;
